slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/SlashingProtectionContextFactory.java

```diff
@@ -38,12 +38,14 @@ public static SlashingProtectionContext create(
 
     // create separate Jdbi instance for pruning operations
     final Jdbi pruningJdbi =
-        DbConnection.createPruningConnection(
-            slashingProtectionParameters.getDbUrl(),
-            slashingProtectionParameters.getDbUsername(),
-            slashingProtectionParameters.getDbPassword(),
-            slashingProtectionParameters.getPruningDbPoolConfigurationFile(),
-            slashingProtectionParameters.isDbConnectionPoolEnabled());
+        slashingProtectionParameters.isPruningEnabled()
+            ? DbConnection.createPruningConnection(
+                slashingProtectionParameters.getDbUrl(),
+                slashingProtectionParameters.getDbUsername(),
+                slashingProtectionParameters.getDbPassword(),
+                slashingProtectionParameters.getPruningDbPoolConfigurationFile(),
+                slashingProtectionParameters.isDbConnectionPoolEnabled())
+            : null;
```

Comment:
its not evident from here, is there a null guard in DbSlashingProtection for pruningJdbi? A better alternative might be to modify DBSlashingProtecting constructor to take `Optional(Jdbi) pruningJdbi` which would make intent more clear that pruningJdbi is optional.

Reply:
Similarly, SlashingProtectionContext should also take `Optional<Jdbi> pruningJdbi` and the usage of it should take appropriate action only if it is present.

Reply:
Done some changes, can you please see if that matches your suggestion?